### ISSUE TITLE
sensor.foxess_solar shows to 15 decimal places

### ISSUE THREAD

**Issue by alexmercerscott** [macxq/foxess-ha#71]
Can the value of "sensor.foxess_solar" be rounded to 2 decimal places? Most of the other figures come in to 1 or 2 decimal places.

Other sensors that also show long decimal places are:

sensor.foxess_feedin
sensor.foxess_bat_discharg
![Screenshot 2022-08-15 at 23 37 25](https://user-images.githubusercontent.com/110939572/184732565-a64bb0b3-0c15-48bf-a3c3-fb8e72fd6b79.jpg)
e

**Comment by sabbie-github** [comment]
Great integration, very happy with it. But bump the 2 decimal request, it would be helpful to have on all the fox sensors.

![image](https://user-images.githubusercontent.com/108624802/185568696-e3e80195-a088-44c6-b74e-dcc215ab91d4.png)
I am just a newbie, playing around with it. In trying to change the above, I have used 2 decimal rounding "|round(2)" in the config yaml - this only changes the sensors that are part of a calculation. 
template:
  - sensor:
      - name: "Solar Power Now"
        unit_of_measurement: "kW"
        state: "{{ (states('sensor.foxess_pv1_power') | float + states('sensor.foxess_pv2_power') | float)|round(2) }}"
Maybe worth noting that this method rounds, and is different from the logic where you want only 2 decimals returned. 
Now I get this, which is better, but doesnt apply to individual sensors. 
![image](https://user-images.githubusercontent.com/108624802/185571298-6bb9267a-86dc-4ca8-8438-87ce5f1bc8c5.png)

**Comment by sabbie-github** [comment]
I now added rounded templates in the config yaml and used these as new sensors to be displayed on the cards. Probably not the most elegant way, but gets the job done

template:
  - sensor:
      - name: "Solar Power Daily"
        unit_of_measurement: "kWh"
        state: "{{ (states('sensor.daily_pv1_sum') | float + states('sensor.daily_pv2_sum') | float)|round(2) }}"
      - name: "Solar Power Now"
        unit_of_measurement: "kW"
        state: "{{ (states('sensor.foxess_pv1_power') | float + states('sensor.foxess_pv2_power') | float)|round(2) }}"
      - name: "FoxESS PV1 Power"
        unit_of_measurement: "kW"
        state: "{{ states('sensor.foxess_pv1_power') | float | round(2) }}"
      - name: "FoxESS PV2 Power"
        unit_of_measurement: "kW"
        state: "{{ states('sensor.foxess_pv2_power') | float | round(2) }}"
      - name: "Battery Discharge Power"
        unit_of_measurement: "kW"
        state: "{{ states('sensor.foxess_bat_discharge_power') | float | round(2) }}"

![image](https://user-images.githubusercontent.com/108624802/185664469-24681013-07ea-4d64-8ec3-d5ab67040455.png)
![image](https://user-images.githubusercontent.com/108624802/185664592-be3b5089-c39d-474d-a73f-daf542b9a7b6.png)

**Comment by Thebig24fan** [comment]
Nice  looking displays  how’s do I add this to my home assistant tried to sort it out but can’t see how only been using ha a few days thanks for any help anyone can give a newbie thanks

**Comment by Davvy80** [comment]
Hi same problem.
How can I fix?

**Comment by sabbie-github** [comment]
@Davvy80 shortening the decimals to 2 can be done by adding sensor templates to the config yaml, which round the sensor values. See post above.

**Comment by Davvy80** [comment]
> @Davvy80 shortening the decimals to 2 can be done by adding sensor templates to the config yaml, which round the sensor values. See post above.

and then I have to select these new sensors?
is your post where there is also the photo?

**Comment by sabbie-github** [comment]
yes, you put the new sensor templates in your config yaml and then use the new sensors in our dashboards

`
template:
  - sensor:
      - name: "FoxESS PV1 Power2"
        unit_of_measurement: "kW"
        state: "{{ '%.2f'%(states('sensor.foxess_pv1_power') | float) }}"
      - name: "FoxESS PV2 Power2"
        unit_of_measurement: "kW"
        state: "{{ '%.2f'%(states('sensor.foxess_pv2_power') | float) }}"
      - name: "Battery Discharge Power"
        unit_of_measurement: "kW"
        state: "{{ '%.2f'%(states('sensor.foxess_bat_discharge_power') | float) }}"

`



the above in the config yaml will create new 2 digit rounded entities to use in your dashboards:
entity: sensor.foxess_pv1_power2
entity: sensor.foxess_pv2_power2
etc
